aea/manager.py

```diff
@@ -524,23 +579,25 @@ def _build_agent_alias(
         agent_name: str,
         agent_overrides: Optional[dict] = None,
         component_overrides: Optional[List[dict]] = None,
+        config: Optional[List[dict]] = None,
     ) -> AgentAlias:
         """Create agent alias for project, with given name and overrided values."""
-        json_config = self._make_config(
-            project.path, agent_overrides, component_overrides
-        )
+        if not config:
+            config = self._make_config(
+                project.path, agent_overrides, component_overrides
+            )
```

Comment:
what if config and overrides are misaligned?

Reply:
The full config is passed only on state recreation. We override if no full config passed here.

Reply:
The problem is, in `add_agent`, a public method, I can now pass inconsistent data. This needs to be avoided. There should be a check that either `agent_overrides` and `component_overrides` are passed (one optionally none) or `config`.

Reply:
Agree. Done.